src/platform/intel/cavs/platform.c

```diff
@@ -313,16 +312,15 @@ int platform_boot_complete(uint32_t boot_message)
 	mailbox_dspbox_write(0, &ready, sizeof(ready));
 
 	/* get any IPC specific boot message and optional data */
-	data = SRAM_WINDOW_HOST_OFFSET(0) >> 12;
-	ipc_boot_complete_msg(&header, &data);
+	ipc_boot_complete_msg(&header, SRAM_WINDOW_HOST_OFFSET(0) >> 12);
```

Comment:
this seems strange - the second parameter of `ipc_boot_complete_msg()` isn't used now any more

Reply:
@lyakh do you mean on CAVS what about other platforms ? We could make this a separate PR if never used ?

Reply:
@lgirdwood I mean this very patch - it modifies `ipc_boot_complete_msg()` to not use its second parameter in both IPC3 and IPC4 implementations and still insists on putting some meaningfully looking data there. That seems strange to me. So, no, it cannot be a separate PR, it very much belongs to this PR

Reply:
@lyakh you're right previously the header part was set inside ipc_boot_complete_msg and data has been set outside of it, in most recent patchset I have corrected the the ipc_boot_complete_msg to keep the previous behavior for IPC3